internal/corazawaf/transaction.go

```diff
@@ -360,6 +361,9 @@ func (tx *Transaction) AddRequestHeader(key string, value string) {
 				tx.variables.requestCookies.AddCS(kl, k, v)
 			}
 		}
+	case "host":
+		// Retrieving from the host header the virtual host to populate serverName
+		tx.variables.serverName.Set(strings.Split(value, ":")[0])
```

Comment:
Hmmm. We are kind of assuming there is only one colon. What happens if someone crafts something like `hostA:hostB:8080`? Coraza will use the first one, but upstream might be confused and use a different one.

Reply:
Not sure if servers / CRS would generally allow a domain with a colon in it to pass through, but definitely an IPV6 address in host would have multiple colons. Good to use `net.SplitHostPort`.